### PR TITLE
100 add a viewmodel for favorite sequences and add favorites to emojirepository

### LINKED ISSUE #100: Add a viewmodel for favorite sequences and add favorites to EmojiRepository
add a viewmodel for favorite sequences that stores the currently being made sequence in memory and also retrieves and stores the sequences from local storage through the EmojiRepository

### PR DESCRIPTION
**Pull request by michaelbrusegard**
IT IS DONE. closes #100. Created new repository instead of adding to EmojiReposiory because of seperation of concerns

### CODE REVIEW

File: app/src/test/kotlin/com/platform/openemoji/favorites/FavoritesRepositoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Test
+    fun testDeleteFavorite() =
+        runBlocking {
+            val favorite = testFavorites[1]
+            favoritesRepository.deleteFavorite(favorite)
+
+            val actualFavorites = favoritesRepository.getFavorites().first()
+
+            assertTrue(!actualFavorites.contains(favorite))
```
Comment (Andreasli02) [inline app/src/test/kotlin/com/platform/openemoji/favorites/FavoritesRepositoryTest.kt]: assertFalse and remove !

File: app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesRepository.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        return if (context != null) {
+            context.dataStore.data
+                .map { preferences ->
+                    val favoritesString = preferences[favoritesPreferencesKey] ?: ""
+                    stringToFavorites(favoritesString)
+                }
+        } else {
+            flowOf(testFavorites ?: emptyList())
+        }
+    }
```
Comment (Andreasli02) [inline app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesRepository.kt]: IDE gives warning: If-Then foldable to '?:' 

You choose if you want to use elvis operator instead

File: app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel(
+    private val favoritesRepository: FavoritesRepository,
+) : ViewModel() {
+    private val _favorites = MutableStateFlow<List<Favorite>?>(null)
+    val favorites = _favorites.asStateFlow()
+
+    private val _currentFavorite = MutableStateFlow<Favorite?>(null)
+    val currentFavorite = _currentFavorite.asStateFlow()
```
Comment (Andreasli02) [inline app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt:18]: Thought that this was supposed to be sequence? I personally think its a better name but you can choose.

File: app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        _currentFavorite.value = favorite.copy(name = name)
+    }
+
+    fun appendToCurrentFavoriteEmojiSequence(emojiCode: String) {
+        val favorite = _currentFavorite.value ?: Favorite("", "")
+        _currentFavorite.value =
+            favorite.copy(
+                emojiSequence = favorite.emojiSequence + emojiCode,
+            )
+    }
```
Comment (Andreasli02) [inline app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt:55]: This name is a little confusing for me.

**Review by Andreasli02 — CHANGES_REQUESTED** [review]
gj :)

File: app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt (hunk 4 above)
Comment (michaelbrusegard) [reply app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt:55]: currentFavorite has an emojiSequence which is a string. The string given to the method (the emojiCode) is appended to the emojiSequence string aka appendToCurrentFavoriteEmojiSequence. It is a little long but it explains what it is

File: app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt (hunk 3 above)
Comment (michaelbrusegard) [reply app/src/main/kotlin/com/platform/openemoji/favorites/FavoritesViewModel.kt:18]: currentFavorite is a favorite object that contains an emojiSequence

**Comment by michaelbrusegard** [comment]
@Andreasli02 everything should be okay now

**Pull request merged**